src/averageAges.js

```diff
@@ -55,7 +73,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = !onlyWithSon
+    ? people.filter(children => (people.find(mother =>
+      children.mother === mother.name)))
+    : people.filter(children => (people.find(mother =>
+      children.mother === mother.name) && children.sex === 'm'));
```

Comment:
you can store result of your filter in `child` variable and then apply `getFilteredGender` if needed